src/component/legend/LegendView.ts

```diff
@@ -226,12 +226,14 @@ class LegendView extends ComponentView {
                     .on('mouseover', curry(dispatchHighlightAction, seriesModel.name, null, api, excludeSeriesId))
                     .on('mouseout', curry(dispatchDownplayAction, seriesModel.name, null, api, excludeSeriesId));
 
-                itemGroup.eachChild(child => {
-                    const ecData = getECData(child);
-                    ecData.seriesIndex = seriesModel.seriesIndex;
-                    ecData.dataIndex = dataIndex;
-                    ecData.ssrType = 'legend';
-                });
+                if (ecModel.ssr) {
```

Comment:
Similarly, can we conditionally register the SSR data getter only when `ssr` is enabled?

https://github.com/apache/echarts/blob/6b8fae82eb5fbd1042b6262dd6b5d39d9adbf373/src/core/echarts.ts#L430-L453